---
output:
  word_document: default
  html_document: default
---
Differential Gene Expression Data of Early Stages of Intestine Regeneration in Holothuria glaberrima
===============================================================================

**Authors:** *Joshua Medina, David Quispe* | 2020

```{r, echo=FALSE, message=FALSE, cache=FALSE}
library(BiocStyle)
knitr::opts_chunk$set(error=FALSE, message=FALSE, warning=FALSE)
```

# Overview

The sea cucumber _Holothuria glaberrima_ has great regenerative capability after internal organs evisceration. In this case, we are analyzing data from mesentery regenerative tissue of 1 and 3 days post evisceration (1- and 3-DPE) in order to further understand the genes that are involved in the early stages of regeneration.

***

# Process

Set the directory to work in and load required libraries.

```{r}

setwd("D:/usuarios/david/Escritorio/Tesis/results/DESeq_protocol/DESeq_newprotocol/GitDESeq/")

suppressMessages(library(DESeq2))
suppressMessages(library(gplots))
suppressMessages(library(RColorBrewer))
suppressMessages(library(EnhancedVolcano))
suppressMessages(library(tidyverse))
suppressMessages(library(ggplot2))
suppressMessages(library(dplyr))
suppressMessages(library(tidyverse))
suppressMessages(library(RColorBrewer))
suppressMessages(library(calibrate))
suppressMessages(library(dichromat))
suppressMessages(library(IDPmisc))
suppressMessages(library(limma))
suppressMessages(library(cowplot))
suppressMessages(library(ggrepel))
suppressMessages(library(pheatmap))
suppressMessages(library(GO.db))
suppressMessages(library(GSEABase))
suppressMessages(library(GOstats))
suppressMessages(library(GOplot))
suppressMessages(library(stargazer))
```


# Cluster analysis

***


* The next analysis comprises count files generated with clustering of similar transcripts using Corset which hierarchically clusters the transcripts based on the distribution of shared reads and expression patterns. 

```{r}
cts<-read.table("data/counts_corset.txt", header = TRUE)
cts<- cts[,c(7,8,9,1,2,3,4,5,6)]
coldata<-read.table("data/pdata_corset.txt", row.names = 1)
```




Gene expression quantification.

```{r}
ddsSTAR_cluster<-DESeqDataSetFromMatrix(countData = cts, colData = coldata, design = ~ condition)
```




Setting Control as the reference level.

```{r}
colData(ddsSTAR_cluster)$condition <- factor(colData(ddsSTAR_cluster)$condition, levels=levels(coldata$condition))
```




Filtering files with low count number.

```{r}
keep_cluster <- rowSums(counts(ddsSTAR_cluster)) >= 30 #Eliminates rows from the samples that have less than 30 counts in the sum of their 3 biological replicates 
ddsSTAR_cluster <- ddsSTAR_cluster[keep_cluster,]
```




Size factor estimation and dispersion. Generation of count table.

```{r}
dds_cluster <- estimateSizeFactors(ddsSTAR_cluster)
dds_cluster <- estimateDispersions(dds_cluster)
dds_cluster <- nbinomWaldTest(dds_cluster)
counts_table_cluster <- counts(dds_cluster, normalized=TRUE)
```



Contrasts between conditions (1DPE vs Normal; 3DPE vs Normal).

```{r}
res_1DPE_2 <- results(dds_cluster, contrast = c("condition", "1DPE", "Control"))
res_1DPE_2 <- res_1DPE_2[order(res_1DPE_2$padj),]

res_1DPEvs3DPE <- results(dds_cluster, contrast = c("condition", "1DPE", "3DPE"))#
res_1DPEvs3DPE <- res_1DPEvs3DPE[order(res_1DPEvs3DPE$padj),]

venn.1DPE<-rbind(res_1DPE_2, res_1DPEvs3DPE)

res_3DPE_2 <- results(dds_cluster, contrast = c("condition", "3DPE", "Control"))
res_3DPE_2 <- res_3DPE_2[order(res_3DPE_2$padj),]

res_3DPEvs1DPE <- results(dds_cluster, contrast = c("condition", "3DPE", "1DPE"))#
res_3DPEvs1DPE <- res_3DPEvs1DPE[order(res_3DPEvs1DPE$padj),]

venn.3DPE <- rbind(res_3DPE_2, res_3DPEvs1DPE)

res_Normvs1DPE <- results(dds_cluster, contrast = c("condition", "Control", "1DPE"))#
res_Normvs1DPE <- res_Normvs1DPE[order(res_Normvs1DPE$padj),]

res_Normvs3DPE <- results(dds_cluster, contrast = c("condition", "Control", "3DPE"))#
res_Normvs3DPE <- res_Normvs3DPE[order(res_Normvs3DPE$padj),]

venn.Norm <- rbind(res_Normvs1DPE, res_Normvs3DPE)
```




Assigning names to the clusters

```{r}
File_1DPE_clus<-as.data.frame(res_1DPE_2)
File_1DPE_clus<-tibble::rownames_to_column(File_1DPE_clus, "ID")
File_3DPE_clus<-as.data.frame(res_3DPE_2)
File_3DPE_clus<-tibble::rownames_to_column(File_3DPE_clus, "ID")
File_3vs1_clus<-as.data.frame(res_3DPEvs1DPE)
File_3vs1_clus<-tibble::rownames_to_column(File_3vs1_clus, "ID")

#Change cluster names with representative Trinity IDs and add protein names
Clusters<-read.delim("data/clusters.txt", header = FALSE, sep = "\t")
GeneNames<-read.csv("data/Genes_infotags.csv")
GeneNames$s_name<- as.character(GeneNames$s_name)
GeneNames$NCBI<- sapply(strsplit(GeneNames$s_name, split = " "), "[", 1)
load("data/Final_NameMap.rda")

File_1DPE_clusName<- File_1DPE_clus %>% left_join(Clusters, by = c("ID"="V2"))
File_3DPE_clusName<- File_3DPE_clus %>% left_join(Clusters, by = c("ID"="V2"))
File_3vs1_clusName<- File_3vs1_clus %>% left_join(Clusters, by = c("ID"="V2"))

tableComb1DPE_clus<-File_1DPE_clusName %>% left_join(Final_NameMap, by=c("V1"="Old_Name"))
tableComb3DPE_clus<-File_3DPE_clusName %>% left_join(Final_NameMap, by=c("V1"="Old_Name"))
tableComb3vs1_clus<-File_3vs1_clusName %>% left_join(Final_NameMap, by=c("V1"="Old_Name"))
Final1DPE_clus<-tableComb1DPE_clus %>% left_join(GeneNames, by=c("value"="q_name"))
Final3DPE_clus<-tableComb3DPE_clus %>% left_join(GeneNames, by=c("value"="q_name"))
Final3vs1_clus<-tableComb3vs1_clus %>% left_join(GeneNames, by=c("value"="q_name"))

```




Eliminating repeats by separating annotated and non annotated transcripts. After separation, distinct() allows to obtain the individual transcripts without losing information. 

```{r}
Final1DPE_clus2<-Final1DPE_clus[is.na(Final1DPE_clus$X10),]
Names1DPE_clus<-Final1DPE_clus[!is.na(Final1DPE_clus$X10),]
Final1DPE_clus3<-distinct(Final1DPE_clus2,Final1DPE_clus2$ID.x, .keep_all = TRUE)
Names1DPE_clus<-Names1DPE_clus[order(Names1DPE_clus$E),]
Names1DPE_clus2<-Names1DPE_clus %>% distinct(Names1DPE_clus$ID.x, .keep_all = TRUE)
pre_finaldis1<-rbind(Names1DPE_clus2[,c(1:7,28,9,34,35,37,41,45,50)],Final1DPE_clus3[,c(1:7,28,9,34,35,37,41,45,50)])
Final1DPE_clus_dis<-distinct(pre_finaldis1, pre_finaldis1$ID.x, .keep_all = TRUE)
Final1DPE_clus_dis<-Final1DPE_clus_dis[,c(-4,-5,-14)]
colnames(Final1DPE_clus_dis)<-c("ID","baseMean","log2FoldChange","pvalue","padj","NCBI","Value","GeneID","LOC","Protein","Synonyms","Description","Other_designations")
Final1DPE_clus_dis<-Final1DPE_clus_dis[order(Final1DPE_clus_dis$padj),]


Final3DPE_clus2<-Final3DPE_clus[is.na(Final3DPE_clus$X10),]
Names3DPE_clus<-Final3DPE_clus[!is.na(Final3DPE_clus$X10),]
Final3DPE_clus3<-distinct(Final3DPE_clus2,Final3DPE_clus2$ID.x, .keep_all = TRUE)
Names3DPE_clus<-Names3DPE_clus[order(Names3DPE_clus$E),]
Names3DPE_clus2<-Names3DPE_clus %>% distinct(Names3DPE_clus$ID.x, .keep_all = TRUE)
pre_finaldis3<-rbind(Names3DPE_clus2[,c(1:7,28,9,34,35,37,41,45,50)],Final3DPE_clus3[,c(1:7,28,9,34,35,37,41,45,50)])
Final3DPE_clus_dis<-distinct(pre_finaldis3, pre_finaldis3$ID.x, .keep_all = TRUE)
Final3DPE_clus_dis<-Final3DPE_clus_dis[,c(-4,-5,-14)]
colnames(Final3DPE_clus_dis)<-c("ID","baseMean","log2FoldChange","pvalue","padj","NCBI","Value","GeneID","LOC","Protein","Synonyms","Description","Other_designations")
Final3DPE_clus_dis<-Final3DPE_clus_dis[order(Final3DPE_clus_dis$padj),]

Final3vs1_clus2<-Final3vs1_clus[is.na(Final3vs1_clus$X10),]
Names3vs1_clus<-Final3vs1_clus[!is.na(Final3vs1_clus$X10),]
Final3vs1_clus3<-distinct(Final3vs1_clus2,Final3vs1_clus2$ID.x, .keep_all = TRUE)
Names3vs1_clus<-Names3vs1_clus[order(Names3vs1_clus$E),]
Names3vs1_clus2<-Names3vs1_clus %>% distinct(Names3vs1_clus$ID.x, .keep_all = TRUE)
pre_finaldis3vs1<-rbind(Names3vs1_clus2[,c(1:7,28,9,34,35,37,41,45,50)],Final3vs1_clus3[,c(1:7,28,9,34,35,37,41,45,50)])
Final3vs1_clus_dis<-distinct(pre_finaldis3vs1, pre_finaldis3vs1$ID.x, .keep_all = TRUE)
Final3vs1_clus_dis<-Final3vs1_clus_dis[,c(-4,-5,-14)]
colnames(Final3vs1_clus_dis)<-c("ID","baseMean","log2FoldChange","pvalue","padj","NCBI","Value","GeneID","LOC","Protein","Synonyms","Description","Other_designations")
Final3vs1_clus_dis<-Final3vs1_clus_dis[order(Final3vs1_clus_dis$padj),]

```

Separate not annotated transcripts

```{r}
Not_annot1DPE<-Final1DPE_clus_dis[is.na(Final1DPE_clus_dis$Protein),]
Not_annot1DPEUp<-Not_annot1DPE[(Not_annot1DPE$padj <= 0.05) & (Not_annot1DPE$log2FoldChange >= 2),]
Not_annot1DPEDown<-Not_annot1DPE[(Not_annot1DPE$padj <= 0.05) & (Not_annot1DPE$log2FoldChange <= -2),]

Not_annot1DPEDown<-!is.na(Not_annot1DPEDown)

write.table(Not_annot1DPEUp$Value[2001:2922], file = "not_annotated/1DPEUp_3.txt", col.names = FALSE, quote = FALSE, row.names = FALSE, sep = "\n")

Not_annot3DPE<-Final3DPE_clus_dis[is.na(Final3DPE_clus_dis$Protein),]
Not_annot3DPEUp<-Not_annot3DPE[(Not_annot3DPE$padj <= 0.05) & (Not_annot3DPE$log2FoldChange >= 2),]
Not_annot3DPEDown<-Not_annot3DPE[(Not_annot3DPE$padj <= 0.05) & (Not_annot3DPE$log2FoldChange <= -2),]

write.table(Not_annot3DPEUp$Value[3001:3329], file = "not_annotated/3DPEUp_4.txt", col.names = FALSE, quote = FALSE, row.names = FALSE, sep = "\n")
```


Generating tables of expressed annotated transcripts

```{r}
MS1DPE<-Final1DPE_clus_dis[order(-Final1DPE_clus_dis$log2FoldChange),]
MS1DPE<-MS1DPE[(!is.na(MS1DPE$Protein)) & (MS1DPE$padj <= 0.05) & (MS1DPE$log2FoldChange >= 2),c(1,10,12,3,5,7)]
MS1DPE<-na.omit(MS1DPE)
MS1DPE<-MS1DPE[-c(1:200),]
MS1DPE$padj<-as.factor(MS1DPE$padj)
stargazer(MS1DPE,type = "html", rownames = FALSE, summary = F, digits = NA, out = "results/Down1DPE.doc")

MS3DPE<-Final3DPE_clus_dis[order(Final3DPE_clus_dis$log2FoldChange),]
MS3DPE<-MS3DPE[(!is.na(MS3DPE$Protein)) & (MS3DPE$padj <= 0.05) & (MS3DPE$log2FoldChange <= -2),c(1,10,12,3,5,7)]
MS3DPE<-na.omit(MS3DPE)
MS3DPE<-MS3DPE[c(1:10),]
MS3DPE$padj<-as.factor(MS3DPE$padj)
stargazer(MS3DPE,type = "html", rownames = FALSE, summary = F, digits = NA, out = "results/Down3DPE.doc")
```




Plotting dispersion estimates.

```{r}
DESeq2::plotDispEsts(dds_cluster,main="Dispersion Plot")
```




Comparison of Raw Data vs normalized counts.

```{r}
conds <- as.vector(coldata$condition)
condcols <- brewer.pal(n = length(unique(conds)), name="Dark2") 
par(mfrow=c(1,2))
barplot(colSums(counts(dds_cluster, normalized=F)), col = condcols[as.factor(conds)], las = 2, cex.names=0.6, main="Raw Counts")
barplot(colSums(counts(dds_cluster, normalized=T)), col=condcols[as.factor(conds)], las=2, cex.names = 0.6, main = "Normalized Counts")
```




PCA plot.

```{r}
vsd_cluster <- varianceStabilizingTransformation(dds_cluster)
DESeq2::plotPCA(vsd_cluster)+theme_bw()
```




Volcano plot of Normal vs 1DPE.

```{r}
with(res_1DPE_2, plot(log2FoldChange, -log10(pvalue), pch=20, main="Time0 vs 1DPE", xlim=c(-30,30)))

with(subset(res_1DPE_2, padj<=0.05 & log2FoldChange>=2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
with(subset(res_1DPE_2, padj<=0.05 & log2FoldChange<=-2), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))

abline(h=-log10(0.05),lty=3)
abline(v=-1,lty=3)
abline(v=1,lty=3)
```

```{r}
volcano_dat1<- Final1DPE_clus_dis[(Final1DPE_clus_dis$padj <= 0.05) & (!is.na(Final1DPE_clus_dis$padj)) & (abs(Final1DPE_clus_dis$log2FoldChange) >= 0),]

keyvals <- ifelse(volcano_dat1$log2FoldChange < -2, 'blue', ifelse(volcano_dat1$log2FoldChange > 2, 'red', 'black'))
keyvals[is.na(keyvals)] <- 'black'
names(keyvals)[keyvals == 'red'] <- 'Up'
names(keyvals)[keyvals == 'black'] <- 'Mid'
names(keyvals)[keyvals == 'blue'] <- 'Down'

tiff(filename='results/volcanoplot1DPE.tiff', width=800, height=600, res = 120)
EnhancedVolcano(volcano_dat1, lab = volcano_dat1$Description, selectLab = c(NA), x = 'log2FoldChange', xlab = bquote(~Log[2]~ "Fold Change"),ylab =bquote(~-Log[10]~ "AdjPval"), y = 'padj', pCutoff = 5e-2, FCcutoff = 2, colCustom = keyvals, transcriptPointSize = 1, colAlpha = 1, legendPosition = 'right', legendLabSize = 12, legendIconSize = 4.0, gridlines.major = TRUE, gridlines.minor = FALSE)
graphics.off()

```



Volcano plot of Normal vs 3DPE.

```{r}
with(res_3DPE_2, plot(log2FoldChange, -log10(pvalue), pch=20, main="Time0 vs 3DPE", xlim=c(-30,30)))

with(subset(res_3DPE_2, padj<=0.05 & log2FoldChange>=2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
with(subset(res_3DPE_2, padj<=0.05 & log2FoldChange<=-2), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))

abline(h=-log10(0.05),lty=3)
abline(v=-1,lty=3)
abline(v=1,lty=3)
```

```{r}
volcano_dat2<- Final3DPE_clus_dis[(Final3DPE_clus_dis$padj <= 0.05) & (!is.na(Final3DPE_clus_dis$padj)) & (abs(Final3DPE_clus_dis$log2FoldChange) >= 0),]

keyvals2 <- ifelse(volcano_dat2$log2FoldChange < -2, 'blue', ifelse(volcano_dat2$log2FoldChange > 2, 'red', 'black'))
keyvals2[is.na(keyvals2)] <- 'black'
names(keyvals2)[keyvals2 == 'red'] <- 'Up'
names(keyvals2)[keyvals2 == 'black'] <- 'Mid'
names(keyvals2)[keyvals2 == 'blue'] <- 'Down'

tiff(filename='results/volcanoplot3DPE.tiff', width=800, height=600, res = 120)
EnhancedVolcano(volcano_dat2, lab = volcano_dat2$Description, selectLab = c(NA), x = 'log2FoldChange', xlab = bquote(~Log[2]~ "Fold Change"),ylab =bquote(~-Log[10]~ "AdjPval"), y = 'padj', pCutoff = 5e-2, FCcutoff = 2, colCustom = keyvals2, transcriptPointSize = 1, colAlpha = 1, legendPosition = 'right', legendLabSize = 12, legendIconSize = 4.0, gridlines.major = TRUE, gridlines.minor = FALSE)
graphics.off()
```

tiff(filename='volcanoplot3DPE.tiff', width=800, height=600, res = 120)
graphics.off()

Venn Diagram

```{r}

contigs_1DPE.venn<-Final1DPE_clus_dis[,c(1,3,5)]
rownames(contigs_1DPE.venn)<-contigs_1DPE.venn$ID
hits_1DPE<-rownames(contigs_1DPE.venn[(contigs_1DPE.venn$padj <= 0.05) & (!is.na(contigs_1DPE.venn$padj)) & (abs(contigs_1DPE.venn$log2FoldChange) >= 2),])


contigs_3DPE.venn<-Final3DPE_clus_dis[,c(1,3,5)]
rownames(contigs_3DPE.venn)<-contigs_3DPE.venn$ID
hits_3DPE<-rownames(contigs_3DPE.venn[(contigs_3DPE.venn$padj <= 0.05) & (!is.na(contigs_3DPE.venn$padj)) & (abs(contigs_3DPE.venn$log2FoldChange) >= 2),])

input_venn <- list(hits_1DPE, hits_3DPE)
names(input_venn) <- c("1DPE", "3DPE")
venn(input_venn)
```

```{r}
hits_1DPE<-rownames(res_1DPE_2[(res_1DPE_2$padj <= 0.05) & (!is.na(res_1DPE_2$padj)) & (abs(res_1DPE_2$log2FoldChange) >= 2),])

hits_3DPE<-rownames(res_3DPE_2[(res_3DPE_2$padj <= 0.05) & (!is.na(res_3DPE_2$padj)) & (abs(res_3DPE_2$log2FoldChange) >= 2),])

input_venn <- list(hits_1DPE, hits_3DPE)
names(input_venn) <- c("1DPE", "3DPE")
venn(input_venn)
```


Venn Diagram upregulated
```{r}
hits_1DPE.UP<-rownames(contigs_1DPE.venn[(contigs_1DPE.venn$padj <= 0.05) & (!is.na(contigs_1DPE.venn$padj)) & (contigs_1DPE.venn$log2FoldChange >= 2),])
hits_3DPE.UP<-rownames(contigs_3DPE.venn[(contigs_3DPE.venn$padj <= 0.05) & (!is.na(contigs_3DPE.venn$padj)) & (contigs_3DPE.venn$log2FoldChange >= 2),])

input_venn.UP <- list(hits_1DPE.UP, hits_3DPE.UP)
names(input_venn.UP) <- c("1DPE", "3DPE")
venn(input_venn.UP)

```

```{r}
keep_singles <- function(v){
  v[!(v %in% v[duplicated(v)])] 
}
hits_1DPE.UP<-rownames(venn.1DPE[(venn.1DPE$padj <= 0.05) & (!is.na(venn.1DPE$padj)) & (venn.1DPE$log2FoldChange >= 2),])
hits_1DPE.UP<-keep_singles(hits_1DPE.UP)
hits_3DPE.UP<-rownames(venn.3DPE[(venn.3DPE$padj <= 0.05) & (!is.na(venn.3DPE$padj)) & (venn.3DPE$log2FoldChange >= 2),])
hits_3DPE.UP<-keep_singles(hits_3DPE.UP)
hits_Norm.UP<-rownames(venn.Norm[(venn.Norm$padj <= 0.05) & (!is.na(venn.Norm$padj)) & (venn.Norm$log2FoldChange >= 2),])
hits_Norm.UP<-keep_singles(hits_Norm.UP)

input_venn.UP <- list(hits_1DPE.UP, hits_3DPE.UP, hits_Norm.UP)
names(input_venn.UP) <- c("1DPE", "3DPE", "Normal")
venn(input_venn.UP)
```



Venn Diagram downregulated
```{r}
hits_1DPE.DOWN<-rownames(contigs_1DPE.venn[(contigs_1DPE.venn$padj <= 0.05) & (!is.na(contigs_1DPE.venn$padj)) & (contigs_1DPE.venn$log2FoldChange <= -2),])
hits_3DPE.DOWN<-rownames(contigs_3DPE.venn[(contigs_3DPE.venn$padj <= 0.05) & (!is.na(contigs_3DPE.venn$padj)) & (contigs_3DPE.venn$log2FoldChange <= -2),])

input_venn.DOWN <- list(hits_1DPE.DOWN, hits_3DPE.DOWN)
names(input_venn.DOWN) <- c("1DPE", "3DPE")
venn(input_venn.DOWN)
```

```{r}
hits_1DPE.DOWN<-rownames(venn.1DPE[(venn.1DPE$padj <= 0.05) & (!is.na(venn.1DPE$padj)) & (venn.1DPE$log2FoldChange <= -2),])
hits_1DPE.DOWN<-keep_singles(hits_1DPE.DOWN)
hits_3DPE.DOWN<-rownames(venn.3DPE[(venn.3DPE$padj <= 0.05) & (!is.na(venn.3DPE$padj)) & (venn.3DPE$log2FoldChange <= -2),])
hits_3DPE.DOWN<-keep_singles(hits_3DPE.DOWN)
hits_Norm.DOWN<-rownames(venn.Norm[(venn.Norm$padj <= 0.05) & (!is.na(venn.Norm$padj)) & (venn.Norm$log2FoldChange <= -2),])
hits_Norm.DOWN<-keep_singles(hits_Norm.DOWN)

input_venn.DOWN <- list(hits_1DPE.DOWN, hits_3DPE.DOWN, hits_Norm.DOWN)
names(input_venn.DOWN) <- c("1DPE", "3DPE", "Normal")
venn(input_venn.DOWN)
```


Table format

```{r}
Stage<-c("Day1","Day1", "Day3", "Day3", "Both", "Both")
Expression<-c("Up-reg","Down-reg","Up-reg","Down-reg","Up-reg","Down-reg")
Counts<-c(1862,2714,1940,2392,1723,2161)

venn<-data.frame(Stage,Expression,Counts)

venn$Stage<- factor(venn$Stage, levels = c("Day1", "Day3", "Both"))
levels(venn$Expression) <-  c("Down", "Up")
levels(venn$Stage) <-  c("Day1", "Day3", "Both")


mycolors <- c("Up" = "red",
                 "Down"  = "blue")  



tiff(filename='results/barplot_venn2.tiff', width=800, height=600, res = 120)
ggplot(data=venn, aes(x=Stage, y = Counts,  fill = reorder(Expression, desc(Expression)))) + geom_bar(stat="identity") + scale_fill_manual(values = mycolors) + labs(y = "Total DEGs", x = "") +geom_text(position = "stack", aes(x=Stage, y = Counts,  label = Counts, hjust = 0.5, vjust=1, fontface="bold")) +theme_minimal() + theme(legend.title = element_blank())
graphics.off()
```



Heatmap of transcription factors

```{r}
tf_1DPE<-Final1DPE_clus_dis[grep("transcription fact", Final1DPE_clus_dis$Protein), ]

tf_1DPE<-distinct(tf_1DPE,tf_1DPE$Description, .keep_all = TRUE)

tf_1DPE<-tf_1DPE[order(-tf_1DPE$log2FoldChange),]

tf_names_1DPE<-tf_1DPE$ID[1:15]

tf_names<-c(tf_names_1DPE,'Cluster-41392.1', 'Cluster-40855.2','Cluster-3567.45288', 'Cluster-36720.3','Cluster-3567.10388','Cluster-33796.0','Cluster-39082.0','Cluster-46175.0','Cluster-3567.41180', 'Cluster-3567.93', 'Cluster-3567.44229','Cluster-3825.2', 'Cluster-42680.1','Cluster-35286.1','Cluster-26511.0','Cluster-38449.0','Cluster-24300.0','Cluster-3567.40893', 'Cluster-51438.0','Cluster-3567.49342','Cluster-43234.1') #aqui se pueden agregar mas transcriptos para que salgan en el heatmap

names_cont1<-Final1DPE_clus_dis[Final1DPE_clus_dis$ID %in% tf_names,c(1,3,5,7,9:12)]

TF_table<-read_delim("data/TF_table.txt", delim = "\t")

names_cont1<-Final1DPE_clus_dis %>% left_join(TF_table, by=c("ID"="ID"))

tf_names<-TF_table$ID

names_cont1<-names_cont1[names_cont1$ID %in% tf_names,c(1,3,5,7,12,16)]

names_cont1<-names_cont1[(names_cont1$padj<=0.05),]

Transcripts<-names_cont1$ID
names_row<-names_cont1$LOC.y

heatmap_dat<-assay(vsd_cluster[c(print(Transcripts))])
rownames(heatmap_dat)<-names_row
heatmap_dat2<-log(heatmap_dat)


heatmap.2(heatmap_dat2, trace='none', cexRow = 0.75,  cexCol = 1.5, las=2, Colv = FALSE, dendrogram='row', key=TRUE, margins=c(2,6), col = bluered(100), srtCol=360, labCol = c("","Normal","","","Day 1","","", "Day 3",""), adjCol = c(0.4,0.5))
```

Generate tiff file of heatmap
```{r}
tiff(filename='heatmapTF_final.tiff', width=800, height=600, res = 120)
heatmap.2(heatmap_dat2, trace='none', cexRow = 0.7,  cexCol = 1.5, las=2, Colv = FALSE, dendrogram='row', key=TRUE, margins=c(2,8), col = bluered(100), srtCol=360, labCol = c("","Normal","","","Day 1","","", "Day 3",""), adjCol = c(0.4,0.5))
graphics.off()
```


```{r}
sampleDists <- dist(t(assay(vsd_cluster)))

sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vsd_cluster$condition, vsd_cluster$type, sep="-")
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)

```



Gene Ontology

```{r}
#Separate all GeneIDs of annotated transcripts
universe_1DPE<-distinct(Final1DPE_clus_dis, Final1DPE_clus_dis$GeneID, .keep_all = TRUE)
#Separate all significant transcripts that are upregulated or downregulated.
genes_1DPEAll_padj0.05<-universe_1DPE[(universe_1DPE$padj <= 0.05) & (!is.na(universe_1DPE$GeneID)),]
genes_1DPEUp_padj0.05<-universe_1DPE[(universe_1DPE$padj <= 0.05) & (!is.na(universe_1DPE$GeneID)) & (universe_1DPE$log2FoldChange >= 2),]
genes_1DPEDown_padj0.05<-universe_1DPE[(universe_1DPE$padj <= 0.05) & (!is.na(universe_1DPE$GeneID)) & (universe_1DPE$log2FoldChange <= -2),]
write(na.omit(universe_1DPE$GeneID), "GO/universe_1DPE.txt", sep = "\n")
write(na.omit(genes_1DPEAll_padj0.05$GeneID), "GO/geneIDs_1DPEAll.txt", sep = "\n")
write(na.omit(genes_1DPEUp_padj0.05$GeneID), "GO/geneIDs_1DPEUp.txt", sep = "\n")
write(na.omit(genes_1DPEDown_padj0.05$GeneID), "GO/geneIDs_1DPEDown.txt", sep = "\n")


universe_3DPE<-distinct(Final3DPE_clus_dis, Final3DPE_clus_dis$GeneID, .keep_all = TRUE)
genes_3DPEAll_padj0.05<-universe_3DPE[(universe_3DPE$padj <= 0.05) & (!is.na(universe_3DPE$GeneID)),]
genes_3DPEUp_padj0.05<-universe_3DPE[(universe_3DPE$padj <= 0.05) & (!is.na(universe_3DPE$GeneID)) & (universe_3DPE$log2FoldChange >= 2),]
genes_3DPEDown_padj0.05<-universe_3DPE[(universe_3DPE$padj <= 0.05) & (!is.na(universe_3DPE$GeneID)) & (universe_3DPE$log2FoldChange <= -2),]
write(na.omit(universe_3DPE$GeneID), "GO/universe_3DPE.txt", sep = "\n")
write(na.omit(genes_3DPEAll_padj0.05$GeneID), "GO/geneIDs_3DPEAll.txt", sep = "\n")
write(na.omit(genes_3DPEUp_padj0.05$GeneID), "GO/geneIDs_3DPEUp.txt", sep = "\n")
write(na.omit(genes_3DPEDown_padj0.05$GeneID), "GO/geneIDs_3DPEDown.txt", sep = "\n")



```


Gene Ontology 3 vs 1 DPE
```{r}
#Separate all GeneIDs of annotated transcripts
universe_3vs1<-distinct(Final3vs1_clus_dis, Final3vs1_clus_dis$GeneID, .keep_all = TRUE)
#Separate all significant transcripts that are upregulated or downregulated.
genes_3vs1Up_padj0.05<-universe_3vs1[(universe_3vs1$padj <= 0.05) & (!is.na(universe_3vs1$GeneID)) & (universe_3vs1$log2FoldChange >= 2),]
genes_3vs1Down_padj0.05<-universe_3vs1[(universe_3vs1$padj <= 0.05) & (!is.na(universe_3vs1$GeneID)) & (universe_3vs1$log2FoldChange <= -2),]
write(na.omit(universe_3vs1$GeneID), "GO/3vs1/universe_3vs1.txt", sep = "\n")
write(na.omit(genes_3vs1Up_padj0.05$GeneID), "GO/3vs1/geneIDs_3vs1Up.txt", sep = "\n")
write(na.omit(genes_3vs1Down_padj0.05$GeneID), "GO/3vs1/geneIDs_3vs1Down.txt", sep = "\n")
```



Make function to produce al tables for revigo after DAVID analysis

```{r}
#Function to generate all the files to submit to Revigo

generate_table<- function(filename) {
  dat<-read.delim(file = filename, header = TRUE, sep = "\t")
  dat2<-str_split(dat$Term, pattern = "[~]")
  dat$Term<-sapply(dat2, "[", 1)
  dat$Description<-sapply(dat2, "[", 2)
  dat<-dat[,c(2,5)]
  file<-gsub(".txt","", filename)
  myfile <- file.path("Revigo", paste0(file, ".txt"))
  write.table(dat, file = myfile, quote = FALSE, row.names = FALSE, sep = "\t")
}


#Function to grab all the files to be applied with the other functions (files need to be in the working directory)

analyze_all<- function(pattern){
  list<-list.files(pattern = pattern, full.names = FALSE)
  for (i in list){
    FilterRev(i)
  }
}


analyze_all("REVIGO")
```

Join tables from Revigo with DAVID results

```{r}
#Function to extract the terms that were simplified with Revigo

FilterRev<-function(filename){
  dat<-read.csv(filename)
  dat2<-dat[(dat$eliminated == 0),1:2]
  file<-gsub(".csv","", filename)
  myfile <- file.path("Revigo_fil", paste0(file, ".txt"))
  write.table(dat2, file = myfile, quote = FALSE, row.names = FALSE, sep = "\t")
}


#Function to join Revigo file with simplified terms and original GO file

JoinGOs<-function(GO, Revigo){
  GOdat<-read.delim(file = GO, header = TRUE, sep = "\t")
  s<-str_split(GOdat$Term, pattern = "[~]")
  GOdat$Term<-sapply(s, "[", 1)
  Revdat<-read.delim(file = Revigo, header = TRUE, sep = "\t")
  final<-GOdat %>% inner_join(Revdat, by=c("Term"="term_ID"))
  file<-gsub(".txt","", GO)
  myfile <- file.path("Final", paste0(file, "_Final", ".txt"))
  write.table(final, file = myfile, quote = FALSE, row.names = FALSE, sep = "\t")
}

JoinGOs("GO_MF3vs1_UP.txt","REVIGO_MF3vs1_UP.txt")


#Function to change category of files

CategorizeBP<-function(filename){
  listBP<-read.delim(file = filename, header = TRUE, sep = "\t")
  listBP$Category<-c("BP")
  fileBP<-gsub(".txt","", filename)
  myfileBP <- file.path("Final", paste0(fileBP, ".txt"))
  write.table(listBP, file = myfileBP, quote = FALSE, row.names = FALSE, sep = "\t")
}


CategorizeCC<-function(filename){
  listCC<-read.delim(file = filename, header = TRUE, sep = "\t")
  listCC$Category<-c("CC")
  fileCC<-gsub(".txt","", filename)
  myfileCC <- file.path("Final", paste0(fileCC, ".txt"))
  write.table(listCC, file = myfileCC, quote = FALSE, row.names = FALSE, sep = "\t")
}

CategorizeMF<-function(filename){
  listMF<-read.delim(file = filename, header = TRUE, sep = "\t")
  listMF$Category<-c("MF")
  fileMF<-gsub(".txt","", filename)
  myfileMF <- file.path("Final", paste0(fileMF, ".txt"))
  write.table(listMF, file = myfileMF, quote = FALSE, row.names = FALSE, sep = "\t")
}

analyze_all<- function(pattern){
  list<-list.files(pattern = pattern, full.names = FALSE)
  for (i in list){
    CategorizeMF(i)
  }
}

analyze_all("GO_MF")

```

After arranging everything, we can join the tables for all, Up and down analysis

```{r}

separate_str<- function(filename) {
  dat<-read.delim(file = filename, header = TRUE, sep = "\t")
  dat2<-str_split(dat$Term, pattern = "[~]")
  dat$Term<-sapply(dat2, "[", 1)
  dat$Description<-sapply(dat2, "[", 2)
  file<-gsub(".txt","", filename)
  myfile <- file.path("Final3", paste0(file, ".txt"))
  write.table(dat, file = myfile, quote = FALSE, row.names = FALSE, sep = "\t")
}




Combinefiles<-function(file1,file2,file3){
  i1<-read.delim(file = file1, header = TRUE, sep = "\t")
  i2<-read.delim(file = file2, header = TRUE, sep = "\t")
  i3<-read.delim(file = file3, header = TRUE, sep = "\t")
  final<-rbind(i1,i2,i3)
  file<-gsub("GO_BP","", file1)
  myfile <- file.path("Final", paste0(file))
  write.table(final, file = myfile, quote = FALSE, row.names = FALSE, sep = "\t")
}

Combinefiles("GO_BP3vs1_UP_Final.txt","GO_CC3vs1_UP_Final.txt","GO_MF3vs1_UP_Final.txt")

```

This is for GOplot
```{r}
wrangle<-function(filename){
  dat<-read.delim(file = filename, header = TRUE, sep = "\t")
  dat2<-dat[c(1,2,14,6,5)]
  colnames(dat2)<-c("Category","ID","Term","Genes","adj_pval")
  myfile <- file.path("FINAL", paste0(filename))
  write.table(dat2, file = myfile, quote = FALSE, row.names = FALSE, sep = "\t")
}

analyze_all<- function(pattern){
  list<-list.files(pattern = pattern, full.names = FALSE)
  for (i in list){
    wrangle(i)
  }
}

analyze_all("Final")
```

Generating Plots from GO
1 DPE


```{r}
GO_1DPE_all<-read.delim("1DPE_All_Final.txt", header = TRUE, sep = "\t")

GO_1DPE_all$GeneRatio<-GO_1DPE_all$Count/GO_1DPE_all$Pop.Hits
Normalizer1All<-max(GO_1DPE_all$Count)/max(GO_1DPE_all$GeneRatio)

GO_1DPE_all2 <- GO_1DPE_all[(GO_1DPE_all$PValue <= 0.01),] %>% group_by(Category) %>% top_n(5, GeneRatio) %>% ungroup() %>% arrange(Category, Count) %>% mutate(Position = n():1)  

tiff(filename='GO_plot.tiff', width=1000, height=800, res = 200)
ggplot(data = GO_1DPE_all2, aes(x = fct_reorder(Description, desc(Position)), y = GeneRatio, fill = Category)) + geom_col(data = GO_1DPE_all2,aes(x = fct_reorder(Description, desc(Position)), y = Count/Normalizer1All)) + scale_y_continuous(sec.axis = sec_axis(trans = ~.*Normalizer1All, name = "Number of genes")) + theme(axis.text.x = element_text(angle = 70, hjust = 1, size = 7), axis.title.y = element_text(size = 8), legend.text = element_text(size = 7), legend.title = element_text(size = 8),legend.key.size =  unit(0.2, "in"), plot.title = element_text(size = 11, hjust = 0.5)) + labs(x = NULL, title = NULL)
graphics.off()
```

```{r}
ggplot(filter(kegg_1DPE_clus, kegg_1DPE_clus$Pvalue<0.01)) +
    geom_point(aes(x = GeneRatio, y = reorder(Pathway, GeneRatio),
               size = Count, color = Pvalue))+ylab("")+theme_bw()+scale_color_gradient2(low = 'blue', mid = 'red')+scale_size_area()
```



Downregulated

```{r}
GO_1DPE_Down<-read.delim("1DPE_Down_Final.txt", header = TRUE, sep = "\t")

GO_1DPE_Down$GeneRatio<-GO_1DPE_Down$Count/GO_1DPE_Down$Pop.Hits
Normalizer1Down<-max(GO_1DPE_Down$Count)/max(GO_1DPE_Down$GeneRatio)

GO_1DPE_Down2 <- GO_1DPE_Down[(GO_1DPE_Down$PValue <= 0.05),] %>% group_by(Category) %>% top_n(5) %>% ungroup() %>% arrange(Category, Count) %>% mutate(Position = n():1)  

tiff(filename='results/GO_1DPEDown.tiff', width=1000, height=800, res = 200)
ggplot(data = GO_1DPE_Down2, aes(x = fct_reorder(Description, desc(Position)), y = GeneRatio, fill = Category)) + geom_col(data = GO_1DPE_Down2, aes(x = fct_reorder(Description, desc(Position)), y = Count/Normalizer1Down)) + scale_y_continuous(sec.axis = sec_axis(trans = ~.*Normalizer1Down, name = "Number of genes")) + theme(axis.text.x = element_text(angle = 70, hjust = 1, size = 7), axis.title.y = element_text(size = 8), legend.text = element_text(size = 7), legend.title = element_text(size = 8),legend.key.size =  unit(0.2, "in"), plot.title = element_text(size = 11, hjust = 0.5)) + labs(x = NULL, title = NULL)
graphics.off()
```

```{r}
GO_1DPE_Up<-read.delim("1DPE_UP_Final.txt", header = TRUE, sep = "\t")

GO_1DPE_Up$GeneRatio<-GO_1DPE_Up$Count/GO_1DPE_Up$Pop.Hits
Normalizer1Up<-max(GO_1DPE_Up$Count)/max(GO_1DPE_Up$GeneRatio)

GO_1DPE_Up2 <- GO_1DPE_Up[(GO_1DPE_Up$PValue <= 0.05),] %>% group_by(Category) %>% top_n(5) %>% ungroup() %>% arrange(Category, Count) %>% mutate(Position = n():1)  

tiff(filename='results/GO_1DPEUp.tiff', width=1000, height=800, res = 200)
ggplot(data = GO_1DPE_Up2, aes(x = fct_reorder(Description, desc(Position)), y = GeneRatio, fill = Category)) + geom_col(data = GO_1DPE_Up2, aes(x = fct_reorder(Description, desc(Position)), y = Count/Normalizer1Up)) + scale_y_continuous(sec.axis = sec_axis(trans = ~.*Normalizer1Up, name = "Number of genes")) + theme(axis.text.x = element_text(angle = 70, hjust = 1, size = 7), axis.title.y = element_text(size = 8), legend.text = element_text(size = 7), legend.title = element_text(size = 8),legend.key.size =  unit(0.2, "in"), plot.title = element_text(size = 11, hjust = 0.5)) + labs(x = NULL, title = NULL)
graphics.off()
```


3 DPE

```{r}
GO_3DPE_all<-read.delim("3DPE_All_Final.txt", header = TRUE, sep = "\t")

GO_3DPE_all$GeneRatio<-GO_3DPE_all$Count/GO_3DPE_all$Pop.Hits
Normalizer3All<-max(GO_3DPE_all$Count)/max(GO_3DPE_all$GeneRatio)

GO_3DPE_all2 <- GO_3DPE_all[(GO_3DPE_all$PValue <= 0.05),] %>% group_by(Category) %>% top_n(5) %>% ungroup() %>% arrange(Category, Count) %>% mutate(Position = n():1)  

ggplot(data = GO_3DPE_all2, aes(x = fct_reorder(Description, desc(Position)), y = GeneRatio, fill = Category)) + geom_col(data = GO_3DPE_all2, aes(x = fct_reorder(Description, desc(Position)), y = Count/Normalizer3All)) + scale_y_continuous(sec.axis = sec_axis(trans = ~.*Normalizer3All, name = "Number of genes")) + theme(axis.text.x = element_text(angle = 70, hjust = 1, size = 7), axis.title.y = element_text(size = 8), legend.text = element_text(size = 7), legend.title = element_text(size = 8),legend.key.size =  unit(0.2, "in"), plot.title = element_text(size = 11, hjust = 0.5)) + labs(x = NULL, title = NULL)
```


```{r}
GO_3DPE_Down<-read.delim("3DPE_Down_Final.txt", header = TRUE, sep = "\t")

GO_3DPE_Down$GeneRatio<-GO_3DPE_Down$Count/GO_3DPE_Down$Pop.Hits
Normalizer3Down<-max(GO_3DPE_Down$Count)/max(GO_3DPE_Down$GeneRatio)

GO_3DPE_Down2 <- GO_3DPE_Down[(GO_3DPE_Down$PValue <= 0.05),] %>% group_by(Category) %>% top_n(5) %>% ungroup() %>% arrange(Category, Count) %>% mutate(Position = n():1)  

tiff(filename='results/GO_3DPEDown.tiff', width=1000, height=800, res = 200)
ggplot(data = GO_3DPE_Down2, aes(x = fct_reorder(Description, desc(Position)), y = GeneRatio, fill = Category)) + geom_col(data = GO_3DPE_Down2, aes(x = fct_reorder(Description, desc(Position)), y = Count/Normalizer3Down)) + scale_y_continuous(sec.axis = sec_axis(trans = ~.*Normalizer3Down, name = "Number of genes")) + theme(axis.text.x = element_text(angle = 70, hjust = 1, size = 7), axis.title.y = element_text(size = 8), legend.text = element_text(size = 7), legend.title = element_text(size = 8),legend.key.size =  unit(0.2, "in"), plot.title = element_text(size = 11, hjust = 0.5)) + labs(x = NULL, title = NULL)
graphics.off()
```

```{r}
GO_3DPE_Down<-read.delim("3vs1_UP_Final.txt", header = TRUE, sep = "\t")

GO_3DPE_Down$GeneRatio<-GO_3DPE_Down$Count/GO_3DPE_Down$Pop.Hits
Normalizer3Down<-max(GO_3DPE_Down$Count)/max(GO_3DPE_Down$GeneRatio)

GO_3DPE_Down2 <- GO_3DPE_Down[(GO_3DPE_Down$PValue <= 0.05),] %>% group_by(Category) %>% top_n(5) %>% ungroup() %>% arrange(Category, Count) %>% mutate(Position = n():1)  

tiff(filename='GO_3vs1_up.tiff', width=1000, height=800, res = 200)
ggplot(data = GO_3DPE_Down2, aes(x = fct_reorder(description, desc(Position)), y = GeneRatio, fill = Category)) + geom_col(data = GO_3DPE_Down2, aes(x = fct_reorder(description, desc(Position)), y = Count/Normalizer3Down)) + scale_y_continuous(sec.axis = sec_axis(trans = ~.*Normalizer3Down, name = "Number of genes")) + theme(axis.text.x = element_text(angle = 70, hjust = 1, size = 7), axis.title.y = element_text(size = 8), legend.text = element_text(size = 7), legend.title = element_text(size = 8),legend.key.size =  unit(0.2, "in"), plot.title = element_text(size = 11, hjust = 0.5)) + labs(x = NULL, title = NULL)
graphics.off()
```


```{r}
GO_3DPE_Up<-read.delim("3DPE_UP_Final.txt", header = TRUE, sep = "\t")

GO_3DPE_Up$GeneRatio<-GO_3DPE_Up$Count/GO_3DPE_Up$Pop.Hits
Normalizer3Up<-max(GO_3DPE_Up$Count)/max(GO_3DPE_Up$GeneRatio)

GO_3DPE_Up2 <- GO_3DPE_Up[(GO_3DPE_Up$PValue <= 0.05),] %>% group_by(Category) %>% top_n(5) %>% ungroup() %>% arrange(Category, Count) %>% mutate(Position = n():1)  

tiff(filename='results/GO_3DPEUp.tiff', width=1000, height=800, res = 200)
ggplot(data = GO_3DPE_Up2, aes(x = fct_reorder(Description, desc(Position)), y = GeneRatio, fill = Category)) + geom_col(data = GO_3DPE_Up2, aes(x = fct_reorder(Description, desc(Position)), y = Count/Normalizer3Up)) + scale_y_continuous(sec.axis = sec_axis(trans = ~.*Normalizer3Up, name = "Number of genes")) + theme(axis.text.x = element_text(angle = 70, hjust = 1, size = 7), axis.title.y = element_text(size = 8), legend.text = element_text(size = 7), legend.title = element_text(size = 8),legend.key.size =  unit(0.2, "in"), plot.title = element_text(size = 11, hjust = 0.5)) + labs(x = NULL, title = NULL)
graphics.off()
```



Pathway Enrichment 1DPE

```{r}
#Filtering IDs with <0.05 of adjusted p-value
keggIDs_1DPE_clus <- genes_1DPEAll_padj0.05$GeneID

#Kegga command from limma package
kegg_1DPE_clus<-kegga(keggIDs_1DPE_clus, species.KEGG = "spu")

#Calculate GeneRation from DE and N
kegg_1DPE_clus$GeneRatio<-kegg_1DPE_clus$DE/kegg_1DPE_clus$N
colnames(kegg_1DPE_clus)<-c("Pathway","N","Count","Pvalue","GeneRatio")

#Using ggplot to visualize results
ggplot(filter(kegg_1DPE_clus, kegg_1DPE_clus$Pvalue<0.01)) +
    geom_point(aes(x = GeneRatio, y = reorder(Pathway, GeneRatio),
               size = Count, color = Pvalue))+ylab("")+theme_bw()+scale_color_gradient2(low = 'blue', mid = 'red')+scale_size_area()
```

Pathway Enrichment Downregulated genes 1DPE

```{r}
#Filtering IDs with <0.05 of adjusted p-value
keggIDs_1DPEDown_clus <- genes_1DPEDown_padj0.05$GeneID

#Kegga command from limma package
kegg_1DPEDown_clus<-kegga(keggIDs_1DPEDown_clus, species.KEGG = "spu")

#Calculate GeneRation from DE and N
kegg_1DPEDown_clus$GeneRatio<-kegg_1DPEDown_clus$DE/kegg_1DPEDown_clus$N
colnames(kegg_1DPEDown_clus)<-c("Pathway","N","Count","Pvalue","GeneRatio")

kegg_1DPEDown_clus<-filter(kegg_1DPEDown_clus, kegg_1DPEDown_clus$Pvalue<0.03)

tiff(filename='results/keggDown_1DPE.tiff', width=1000, height=800, res = 150)
ggplot(kegg_1DPEDown_clus) + geom_col(aes(x=GeneRatio, y= reorder(Pathway, GeneRatio), fill=Pvalue)) + scale_fill_viridis_c(direction = -1,end = .7, option = "plasma") + ylab("") + theme_bw()
graphics.off()
```



Pathway Enrichment Upregulated genes 1DPE 

```{r}
#Filtering IDs with <0.05 of adjusted p-value
keggIDs_1DPEUp_clus <- genes_1DPEUp_padj0.05$GeneID

#Kegga command from limma package
kegg_1DPEUp_clus<-kegga(keggIDs_1DPEUp_clus, species.KEGG = "spu")

#Calculate GeneRation from DE and N
kegg_1DPEUp_clus$GeneRatio<-kegg_1DPEUp_clus$DE/kegg_1DPEUp_clus$N
colnames(kegg_1DPEUp_clus)<-c("Pathway","N","Count","Pvalue","GeneRatio")

kegg_1DPEUp_clus<-filter(kegg_1DPEUp_clus, kegg_1DPEUp_clus$Pvalue<0.05)

tiff(filename='results/keggUp_1DPE.tiff', width=1000, height=800, res = 150)
ggplot(kegg_1DPEUp_clus) + geom_col(aes(x=GeneRatio, y= reorder(Pathway, GeneRatio), fill=Pvalue)) + scale_fill_viridis_c(direction = -1,end = .7, option = "plasma") + ylab("") + theme_bw()
graphics.off()
```



Pathway Enrichment 3DPE

```{r}
#Filtering IDs with <0.05 of adjusted p-value
keggIDs_3DPE_clus <- genes_3DPEAll_padj0.05$GeneID

#Kegga command from limma package
kegg_3DPE_clus<-kegga(keggIDs_3DPE_clus, species.KEGG = "spu")

#Calculate GeneRation from DE and N
kegg_3DPE_clus$GeneRatio<-kegg_3DPE_clus$DE/kegg_3DPE_clus$N
colnames(kegg_3DPE_clus)<-c("Pathway","N","Count","Pvalue","GeneRatio")

#Using ggplot to visualize results
ggplot(filter(kegg_3DPE_clus, kegg_3DPE_clus$Pvalue<0.01)) +
    geom_point(aes(x = GeneRatio, y = reorder(Pathway, GeneRatio),
               size = Count, color = Pvalue))+ylab("")+theme_bw()+scale_color_gradient2(low = 'blue', mid = 'red')+scale_size_area()
```




Pathway enrichment analysis downregulated 3DPE

```{r}
#Filtering IDs with <0.05 of adjusted p-value
keggIDs_3DPEDown_clus <- genes_3DPEDown_padj0.05$GeneID

#Kegga command from limma package
kegg_3DPEDown_clus<-kegga(keggIDs_3DPEDown_clus, species.KEGG = "spu")

#Calculate GeneRation from DE and N
kegg_3DPEDown_clus$GeneRatio<-kegg_3DPEDown_clus$DE/kegg_3DPEDown_clus$N
colnames(kegg_3DPEDown_clus)<-c("Pathway","N","Count","Pvalue","GeneRatio")

kegg_3DPEDown_clus<-filter(kegg_3DPEDown_clus, kegg_3DPEDown_clus$Pvalue<0.05)

tiff(filename='results/keggDown_3DPE.tiff', width=1000, height=800, res = 150)
ggplot(kegg_3DPEDown_clus) + geom_col(aes(x=GeneRatio, y= reorder(Pathway, GeneRatio), fill=Pvalue)) + scale_fill_viridis_c(direction = -1,end = .7, option = "plasma") + ylab("") + theme_bw()
graphics.off()
```




Pathway enrichment analysis upregulated 3DPE

```{r}
#Filtering IDs with <0.05 of adjusted p-value
keggIDs_3DPEUp_clus <- genes_3DPEUp_padj0.05$GeneID

#Kegga command from limma package
kegg_3DPEUp_clus<-kegga(keggIDs_3DPEUp_clus, species.KEGG = "spu")

#Calculate GeneRation from DE and N
kegg_3DPEUp_clus$GeneRatio<-kegg_3DPEUp_clus$DE/kegg_3DPEUp_clus$N
colnames(kegg_3DPEUp_clus)<-c("Pathway","N","Count","Pvalue","GeneRatio")

kegg_3DPEUp_clus<-filter(kegg_3DPEUp_clus, kegg_3DPEUp_clus$Pvalue<0.05)

tiff(filename='results/keggUp_3DPE.tiff', width=1000, height=800, res = 150)
ggplot(kegg_3DPEUp_clus) + geom_col(aes(x=GeneRatio, y= reorder(Pathway, GeneRatio), fill=Pvalue)) + scale_fill_viridis_c(direction = -1,end = .7, option = "plasma") + ylab("") + theme_bw()
graphics.off()
```


***


```{r}
counts_df<-rownames_to_column(as.data.frame(counts_table_cluster), var = "ID")
Data_DPE<-Final1DPE_clus_dis %>% left_join(Final3DPE_clus_dis, by=c("ID"="ID"))
Data_DPE<-Data_DPE[,c(1,3,13)]
rownames(Data_DPE)<-Data_DPE$ID
Data_DPE<-Data_DPE[,-1]
colnames(Data_DPE)<-c("1DPE","3DPE")
```


```{r}
nSets=3
setLabels<-c("Normal","1 DPE","3 DPE")
shortLabels<-c("Normal","1DPE","3DPE")
multiExpr = vector(mode = "list", length = nSets)

multiExpr[[1]] = list(data = as.data.frame(t(counts_df[,c(2:4)])))
names(multiExpr[[1]]$data) = counts_df$ID
rownames(multiExpr[[1]]$data) = names(counts_df)[2:4]
multiExpr[[2]] = list(data = as.data.frame(t(counts_df[,(5:7)])))
names(multiExpr[[2]]$data) = counts_df$ID
rownames(multiExpr[[2]]$data) = names(counts_df)[5:7]
multiExpr[[3]] = list(data = as.data.frame(t(counts_df[,(8:10)])))
names(multiExpr[[2]]$data) = counts_df$ID
rownames(multiExpr[[2]]$data) = names(counts_df)[8:10]
exprSize=checkSets(multiExpr)
exprSize

```

```{r}
gsg = goodSamplesGenesMS(multiExpr, verbose = 3)
gsg$allOK
```


```{r}
data(CIT_BLCA_EXP,HumanTF,CIT_BLCA_Subgroup)
```


```{r}
Samples<-c("Norm17A","Norm17B","Norm_19","Day1A","Day1B","Day1C","Day3A","Day3B","Day3C")
Groups<-c("Normal","Normal","Normal","Day1","Day1","Day1","Day3","Day3","Day3")
Subgroups_Counts<-as.data.frame(Groups,Samples)
```

```{r}
grn=hLICORN(Data_DPE, TFlist = tf_names)
influence=regulatorInfluence(grn,counts_table_cluster)
coregs=coregulators(grn)
display(grn,counts_table_cluster,influence,clinicalData = Subgroups)
```


```{r}
TF_table<-read_delim("data/TF_table.txt", delim = "\t")
```


```{r}
tf_1DPE<-Final1DPE_clus_dis[grep("transcription fact", Final1DPE_clus_dis$Protein), ]

tf_1DPE<-distinct(tf_1DPE,tf_1DPE$Description, .keep_all = TRUE)

tf_1DPE<-tf_1DPE[order(-tf_1DPE$log2FoldChange),]

tf_names_1DPE<-tf_1DPE$ID[1:15]

tf_names<-c(tf_names_1DPE,'Cluster-41392.1', 'Cluster-40855.2','Cluster-3567.45288', 'Cluster-36720.3','Cluster-3567.10388','Cluster-33796.0','Cluster-39082.0','Cluster-46175.0','Cluster-3567.41180', 'Cluster-3567.93', 'Cluster-3567.44229','Cluster-3825.2', 'Cluster-42680.1','Cluster-35286.1','Cluster-26511.0','Cluster-38449.0','Cluster-24300.0','Cluster-3567.40893', 'Cluster-51438.0','Cluster-3567.49342','Cluster-43234.1') #aqui se pueden agregar mas transcriptos para que salgan en el heatmap

names_cont1<-Final1DPE_clus_dis[Final1DPE_clus_dis$ID %in% tf_names,c(1,3,5,7,9:12)]

names_cont1<-names_cont1[(names_cont1$padj<=0.05) & (abs(names_cont1$log2FoldChange) >= 2),]

Transcripts<-names_cont1$ID
names_row<-names_cont1$LOC

heatmap_dat<-assay(vsd_cluster[c(print(Transcripts))])
rownames(heatmap_dat)<-names_row
heatmap_dat2<-log(heatmap_dat)

heatmap.2(heatmap_dat2, trace='none', cexRow = 0.9,  cexCol = 1.5, las=2, Colv = FALSE, dendrogram='row', key=TRUE, margins=c(2,8), col = bluered(100), srtCol=360, labCol = c("","Normal","","","Day 1","","", "Day 3",""), adjCol = c(0.4,0.5))
```



CoRegNet Data

```{r}
tf_1DPE<-Final1DPE_clus_dis[grep("transcription fact", Final1DPE_clus_dis$Protein), ]

tf_1DPE<-distinct(tf_1DPE,tf_1DPE$Description, .keep_all = TRUE)

tf_1DPE<-tf_1DPE[order(-tf_1DPE$log2FoldChange),]

tf_names_1DPE<-tf_1DPE$ID

tf_names<-c(tf_names_1DPE,'Cluster-41392.1', 'Cluster-40855.2','Cluster-3567.45288', 'Cluster-36720.3','Cluster-3567.10388','Cluster-33796.0','Cluster-39082.0','Cluster-46175.0','Cluster-3567.41180', 'Cluster-3567.93', 'Cluster-3567.44229','Cluster-3825.2', 'Cluster-42680.1','Cluster-35286.1','Cluster-26511.0','Cluster-38449.0','Cluster-24300.0','Cluster-3567.40893', 'Cluster-51438.0','Cluster-3567.49342','Cluster-43234.1')

TF_table<-read_delim("data/TF_table.txt", delim = "\t")

Data_DPE<-Final1DPE_clus_dis %>% left_join(Final3DPE_clus_dis, by=c("ID"="ID"))

names_cont1<-Data_DPE %>% left_join(TF_table, by=c("ID"="ID"))

Data_DPE<-names_cont1[!is.na(names_cont1$GeneID.x),c(1,3,15,8,28)]

Data_DPE$ID <- ifelse(is.na(Data_DPE$LOC), Data_DPE$ID, Data_DPE$LOC)

rownames(Data_DPE)<-make.names(Data_DPE$ID, unique = TRUE)

Data_DPE<-Data_DPE[,c(-1,-4,-5)]

colnames(Data_DPE)<-c("1DPE","3DPE")


#Interactions
String_int<-read_delim("co-reg/STRING_int.txt", delim = "\t", col_names = FALSE)

test<-Data_DPE[rownames(Data_DPE) %in% String_int$X2,]

write.table(String_int, "co-reg/STRING_int.txt", sep = "\t")
```

